Mage/src/main/java/mage/abilities/effects/common/continuous/GainAbilityTargetEffect.java

```diff
@@ -44,6 +43,8 @@ public GainAbilityTargetEffect(Ability ability, Duration duration, String rule)
     public GainAbilityTargetEffect(Ability ability, Duration duration, String rule, boolean useOnCard) {
         super(duration, Layer.AbilityAddingRemovingEffects_6, SubLayer.NA, ability.getEffects().getOutcome(ability, Outcome.AddAbility));
         this.ability = ability;
+        this.ability = copyAbility();
```

Comment:
Is this an overrideable method in the constructor? That's a potential concern

Reply:
I'll make the function private, it has no business being public or protected.